js/src/components/free-listings/setup-free-listings/index.js

```diff
@@ -115,7 +115,29 @@ const SetupFreeListings = ( {
 		} else if ( change.name === 'shipping_country_times' ) {
 			onShippingTimesChange( values.shipping_country_times );
 		} else if ( settingsFieldNames.includes( change.name ) ) {
-			onSettingsChange( getSettings( values ) );
+			// The value of `shipping_time` option is determined by the value of `shipping_rate` option.
+			// So if the current form change is considered it needs to change `shipping_time` as well,
+			// it schedules the processing with `formPropsDelegateeRef` and also skips the call of
+			// `onSettingsChange` this time, and lets the call of `onSettingsChange` be triggered
+			// when the form change of `shipping_time` happens.
+			let shouldTriggerOnChange = true;
+
+			if ( change.name === 'shipping_rate' ) {
+				// When shipping rate is 'manual', shipping time should be 'manual' as well;
+				// When shipping rate is 'automatic' or 'flat', shipping time should be 'flat'.
+				const nextValue = change.value === 'manual' ? 'manual' : 'flat';
+
+				if ( nextValue !== values.shipping_time ) {
+					shouldTriggerOnChange = false;
```

Comment:
:question: 
I found it counterintuitive, that if the next value is different we do not trigger `onChange`, but where it is the same (nothing changed) we do trigger it.

Reply:
Sorry, it's explained above

>             // it schedules the processing with `formPropsDelegateeRef` and also skips the call of
> 			// `onSettingsChange` this time, and lets the call of `onSettingsChange` be triggered
> 			// when the form change of `shipping_time` happens.